chains/evm/contracts/onRamp/ExecutorOnRamp.sol

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.24;
+
+import {IExecutorOnRamp} from "../interfaces/IExecutorOnRamp.sol";
+
+import {Client} from "../libraries/Client.sol";
+
+import {Ownable2StepMsgSender} from "@chainlink/contracts/src/v0.8/shared/access/Ownable2StepMsgSender.sol";
+import {EnumerableSet} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v5.0.2/contracts/utils/structs/EnumerableSet.sol";
+
+/// @notice The ExecutorOnRamp configures executor fees, destination support, and CCV limits.
+contract ExecutorOnRamp is Ownable2StepMsgSender, IExecutorOnRamp {
+  using EnumerableSet for EnumerableSet.AddressSet;
+  using EnumerableSet for EnumerableSet.UintSet;
+
+  error ExceedsMaxPossibleCCVs(uint256 provided, uint256 max);
+  error ExceedsMaxRequiredCCVs(uint256 provided, uint256 max);
+  error InvalidCCV(address ccv);
+  error InvalidConfig();
+  error InvalidDestChain(uint64 destChainSelector);
+  error InvalidExtraArgsVersion(bytes4 provided);
+
+  event ConfigSet(DynamicConfig dynamicConfig);
+  event DestChainAdded(uint64 indexed destChainSelector);
+  event DestChainRemoved(uint64 indexed destChainSelector);
+  event CCVAdded(address indexed ccv);
+  event CCVRemoved(address indexed ccv);
+
+  /// @dev Struct that defines the dynamic configuration.
+  // solhint-disable-next-line gas-struct-packing
+  struct DynamicConfig {
+    // Address that quotes a fee for each message
+    address feeQuoter;
```

Comment:
Generally we don't add things that are not used in the PR, like feeQuoter/feeAggegator. Every PR should be a complete unit of logic that is shippable.